test/modules/sungeun/init/printModuleInitOrder.na-none.good

```diff
@@ -1,16 +1,6 @@
 Initializing Modules:
    ChapelStandard
-    CString
-    Bytes
-     ByteBufferHelpers
-     BytesStringCommon
-      String
```

Comment:
Good reminder, I'll run the other .good file.

As for why these particular modules are no longer being loaded - the program itself only contains a use statement to a program that defines a single variable (of bool type).  I tend to think of `writeln` as what triggers the use of strings in simple programs.  ChapelBase is mostly functions, with very few module level symbols that would end up in an initialization function so perhaps none of the module level symbols are being used in a basic program?

Reply:
Just for our sanity, would you try `--printModuleInitOrder` with some Hello World type programs? Say, one with just a string literal in a variable and nothing else? And make sure that things like String are initialized if they are definitely used?

I would think that a program containing a string literal would need `String` because the compiler emits calls to string initialization functions. I have a hard time believing that it's possible to write a Chapel program that doesn't have any string literals (I'd imagine there are some in the standard modules, right?).

Reply:
Hmm, I'm not seeing it there either.  Looking at the generated C code, `chpl__init_String` and `chpl__init_ChapelBase` do get called by `chpl__init_Atomics` (which does print out), and the flag to ensure it doesn't get called more than once is only set by `chpl__init_preInit` and the module initialization function itself.  I'll dig more into why that isn't sufficient for it to get printed out, thanks for noticing it!

Reply:
I figured it out.  This is because the module PrintModuleInitOrder got a use of ChapelBase added - as a result, the module initialization function is getting triggered when PrintModuleInitOrder sets itself up (which occurs *prior* to when the flag to print the initialized module names is set), and then ChapelBase and its dependencies mark themselves as "don't visit my init function again", so by the time everything else gets initialized, they get skipped.

Reply:
How about we change printModuleInitOrder into a param-only flag? I think it's better not to add exec-time flags to every program unless they're really useful to users.

Reply:
This is what the new output looks like.  We always evaluate our dependencies first, so I think to get this to print out nicely, we might need to move it inside the compiler.  Or at least the "Initializing Modules" portion.  Additionally, since PrintModuleInitOrder impacts the initialization order by being used, I'm not sure it provides value as a module after this change.

```
  PrintModuleInitOrder
   ChapelBase
    HaltWrappers
    String
     CString
     SysCTypes
     StringCasts
     ByteBufferHelpers
     BytesStringCommon
      Bytes
     SysBasic
Initializing Modules:
   ChapelStandard
    OwnedObject
    Atomics
     MemConsistency
    ChapelThreads
    ChapelTuple
    ChapelRange
    ChapelTaskDataHelp
    LocaleModel
     LocaleModelHelpFlat
      LocaleModelHelpSetup
       ChapelLocale
       DefaultRectangular
        DSIUtil
        ChapelArray
         ChapelDebugPrint
         Sort
          List
           ChapelLocks
          MSBRadixSort
           RadixSortHelp
           ShellSort
          QuickSort
           InsertionSort
          BubbleSort
          HeapSort
          BinaryInsertionSort
          MergeSort
          SelectionSort
        ChapelDistribution
         LinkedLists
        SysError
        IO
         FormattedIO
          Regexp
        DefaultAssociative
        ExternalArray
        RangeChunk
       ChapelNumLocales
       Sys
      LocaleModelHelpRuntime
     LocaleModelHelpMem
    LocalesArray
    ChapelIO
    LocaleTree
    ChapelTaskTable
    MemTracking
    ChapelUtil
    ChapelError
    ChapelTaskData
    Assert
    AlignedTSupport
    CPtr
   printModuleInitOrder
    moduleInitTest
```

Reply:
Thanks for trying it out. Let's move your draft patch & discussion about what to do with this feature to an issue for the moment, and this PR can go ahead with the odd impact.

Reply:
I opened #14339